datafusion/expr/src/expr_rewriter.rs

```diff
@@ -519,6 +495,49 @@ pub fn unnormalize_cols(exprs: impl IntoIterator<Item = Expr>) -> Vec<Expr> {
     exprs.into_iter().map(unnormalize_col).collect()
 }
 
+struct RewriterAdapter<F> {
+    f: F,
+}
+
+impl<F> ExprRewriter for RewriterAdapter<F>
+where
+    F: FnMut(Expr) -> Result<Expr>,
+{
+    fn mutate(&mut self, expr: Expr) -> Result<Expr> {
```

Comment:
Can we let `ExprRewriter::mutate` return `Result<Option<E>>` in the future, just like what we've done for `LogicalPlan::try_optimize`?

Reply:
That is an interesting idea. What usecase do you have in mind? 

Since `mutate` takes the expr by ownership (rather than by reference), I think returning `expr` would be the same as not changing the original `expr` (perhaps the same as returning `None`?)

However, I do wonder what actually happens when rewriting Boxed exprs 🤔 maybe they are still copied